### PR TITLE
[16.0][IMP] l10n_br_nfse_focus: add field percentual_aliquota_relativa_municipio in payload for nfse nacional, add field codigo_municipio_incidencia in payload for nfse nacional - FWP #4368 #4369

### PR DESCRIPTION
**Pull request by CristianoMafraJunior**
FWP #4368 #4369

### CODE REVIEW

**Comment by CristianoMafraJunior** [comment]
Eu acho que esse commit aqui https://github.com/odoo/odoo/commit/7763099c91383745464f55b3a8ef13eb0d7f31f2 ta quebrando os testes no l10n_br_account

```
2026-02-11 12:15:14,215 563 ERROR odoo odoo.addons.l10n_br_account.tests.test_multi_localizations_invoice: FAIL: MultiLocalizationsInvoice.test_force_out_invoice_create_refund
Traceback (most recent call last):
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_account/tests/test_multi_localizations_invoice.py", line 127, in test_force_out_invoice_create_refund
    return super().test_out_invoice_create_refund()
  File "/opt/odoo/addons/account/tests/test_account_move_out_invoice.py", line 1810, in test_out_invoice_create_refund
    self.assertInvoiceValues(reverse_move, [
  File "/opt/odoo/addons/account/tests/common.py", line 526, in assertInvoiceValues
    self.assertRecordValues(move, [expected_move_values])
  File "/opt/odoo/odoo/tests/common.py", line 616, in assertRecordValues
    self.fail('\n'.join(errors))
AssertionError: The records and expected_values do not match.

==== Differences at index 0 ====
--- 

+++ 

@@ -1 +1 @@

-partner_bank_id:False
+partner_bank_id:49
```

**Comment by felipemotter** [comment]
Precisa dessa pr para liberar o ci... https://github.com/OCA/bank-payment/pull/1554

**Comment by felipemotter** [comment]
mesmo com a https://github.com/OCA/bank-payment/pull/1554 que acabou de entrar nao foi suficiente... abri uma outra PR que agora sim corrige definitivamente. Segue: https://github.com/OCA/bank-payment/pull/1555

**Comment by antoniospneto** [comment]
@CristianoMafraJunior precisa dar rebase aqui tbm

**Comment by CristianoMafraJunior** [comment]
> @CristianoMafraJunior precisa dar rebase aqui tbm

done

**Review by antoniospneto — APPROVED** [review]

**Comment by marcelsavegnago** [comment]
ping @CristianoMafraJunior

**Comment by CristianoMafraJunior** [comment]
> ping @CristianoMafraJunior

donee

**Review by marcos-mendez — CHANGES_REQUESTED** [review]
**Automated Review -- Tests Failed**

### 1. **Root Cause**

The test failure is caused by a **database connection error** during test execution, not directly from the code changes in the PR. However, the changes introduce a new field `aliquota_iss` being added to the payload for NFSe (Nota Fiscal de Serviço Eletrônica) generation, which may not be properly handled in existing tests or models, leading to potential issues in test setup or data initialization.

---

### 2. **Suggested Fix**

The new field `aliquota_iss` must be defined in the corresponding Odoo model (`nfse_nacional.py`) or its related model to avoid runtime errors when the field is referenced in views or logic. Specifically:

- **File:** `l10n_br_nfse_focus/models/nfse_nacional.py`
- **Line:** Add the field `aliquota_iss` to the model definition (likely in the class that handles NFSe data)
- **Suggestion:** Add a field definition like:
  ```python
  aliquota_iss = fields.Float(string="Alíquota ISS")
  ```

Also, ensure that the field is properly initialized or mapped in `_prepare_service_basic_nacional` and `_prepare_payload_nacional` methods to avoid KeyError or missing field errors in test environments.

---

### 3. **Additional Code Issues**

- **Potential KeyError in `aliquota_iss` assignment:** The code assumes `service_info.get("aliquota", 0)` exists. If `service_info` is malformed or missing the key, it could cause an error. Consider using a default fallback or validation.
  
  ```python
  aliquota = service_info.get("aliquota", 0)
  ```

- **Redundant conditional in `codigo_municipio_incidencia`:** The conditional block is correct but could be simplified for readability:
  ```python
  if service_basic.get("tributacao_iss") == "1":
      payload.update({
          "codigo_municipio_incidencia": service_basic["codigo_municipio_prestacao"]
      })
  ```

---

### 4. **Test Improvements**

To improve test coverage and prevent regressions:

- **Add a TransactionCase** to test the new `aliquota_iss` field in both `_prepare_service_basic_nacional` and `_prepare_payload_nacional` methods.
- **Test edge cases**:
  - When `aliquota` is `None` or `0`
  - When `tributacao_iss` is `"1"` vs other values
  - When `codigo_municipio_incidencia` is or isn't included in the payload
- **Use tagged tests** to ensure that the NFSe payload generation logic is tested under different configurations (e.g., with and without municipal incidence).

**Example test snippet (in TransactionCase):**

```python
def test_prepare_service_basic_nacional_with_aliquota(self):
    service_info = {
        "aliquota": 0.05,
        "valor": 1000.0,
        # other fields
    }
    result = self.env['l10n_br_nfse.focus'].with_context()._prepare_service_basic_nacional(service_info)
    self.assertEqual(result['aliquota_iss'], 5.0)
```

This ensures that the new field is correctly computed and integrated into the payload.

--- 

✅ **Conclusion:** The PR introduces a new field that must be defined in the model to prevent runtime errors. The test failure is likely due to an environment/database issue, but the code changes should be validated with proper test coverage to ensure robustness.

---
⏰ This PR has been open for **32 days**.

> _Every ignored PR is a contributor who might not come back. Review time matters. ([OCA Aging Report](https://git.pop.coop/pop/oca-reviewer))_

---
### Reciprocal Review Request



Hi everyone! I found some test failures on this PR and left detailed feedback above. I am happy to discuss or help debug. In the meantime, if any of you get a chance, I would appreciate a look at my open PR(s):

**My open PRs across OCA:**
- [server-tools#3554 [MIG] datetime_formatter: Migration to 18.0](https://github.com/OCA/server-tools/pull/3554)
- [server-tools#3548 [18.0][MIG] base_kanban_stage: Migration to 18.0](https://github.com/OCA/server-tools/pull/3548)
- [hr-attendance#262 [16.0][ADD] Hr_attendance_idsecure: iDSecure (ControliD) attendance integration](https://github.com/OCA/hr-attendance/pull/262)
- [stock-logistics-workflow#2276 [16.0][ADD] stock_move_line_devaluation](https://github.com/OCA/stock-logistics-workflow/pull/2276)
- [stock-logistics-workflow#2275 [16.0][ADD] Stock move line analytic account](https://github.com/OCA/stock-logistics-workflow/pull/2275)
- [stock-logistics-workflow#2268 [16.0][ADD] stock_move_line_picking_partner](https://github.com/OCA/stock-logistics-workflow/pull/2268)
- [purchase-workflow#2694 [16.0][IMP]Purchase workflow added to review state & exception fix](https://github.com/OCA/purchase-workflow/pull/2694)

Reviewing each other's work helps the whole community move forward. Thank you!

---
_Environment via [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer): Minikube + K8s Job + oca-ci/py3.10-odoo16.0 | Odoo 16.0_
_Automated review by [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer) + qwen3-coder:30b_

**Review by antoniospneto — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]
LGTM
